relayer/relayer/src/handlers/azero.rs

```diff
@@ -0,0 +1,267 @@
+                    request_hash,
+                    committee_id.into(),
+                    dest_token_address,
+                    amount.into(),
+                    dest_receiver_address,
+                    request_nonce.into(),
+                );
+
+                info!(
+                "Sending tx with request nonce {} to the Ethereum network and waiting for {} confirmations.",
+                request_nonce,
+                eth_tx_min_confirmations
+            );
+
+                // This shouldn't fail unless there is something wrong with our config.
+                // NOTE: this does not check whether the actual tx reverted on-chain. Reverts are only checked on dry-run.
+                let receipt = call
+                    .gas(config.eth_gas_limit)
+                    .nonce(eth_connection.inner().next())
+                    .send()
+                    .await?
+                    .confirmations(*eth_tx_min_confirmations)
+                    .retries(*eth_tx_submission_retries)
+                    .await?
+                    .ok_or(AlephZeroEventHandlerError::TxNotPresentInBlockOrMempool)?;
+
+                let tx_hash = receipt.transaction_hash;
+                let tx_status = receipt.status;
+
+                // Check if the tx reverted.
+                if tx_status == Some(U64::from(0)) {
+                    warn!(
+                    "Tx with nonce {request_nonce} has been sent to the Ethereum network: {tx_hash:?} but it reverted."
+                );
+                    return Err(AlephZeroEventHandlerError::EthContractReverted);
+                }
+
+                info!("Tx with nonce {request_nonce} has been sent to the Ethereum network: {tx_hash:?} and received {eth_tx_min_confirmations} confirmations.");
+
+                wait_for_eth_tx_finality(eth_connection, tx_hash).await?;
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Error)]
+#[error(transparent)]
+#[non_exhaustive]
+pub enum AlephZeroEventsHandlerError {
+    #[error("events ack receiver dropped")]
+    EventsAckReceiverDropped,
+
+    #[error("broadcast receive error")]
+    BroadcastReceive(#[from] broadcast::error::RecvError),
+
+    #[error("broadcast send error")]
+    BroadcastSend(#[from] broadcast::error::SendError<CircuitBreakerEvent>),
+
+    #[error("Task join error")]
+    Join(#[from] JoinError),
+}
+
+pub struct AlephZeroEventsHandler;
+
+impl AlephZeroEventsHandler {
+    pub async fn run(
+        config: Arc<Config>,
+        eth_signed_connection: Arc<SignedEthConnection>,
+        mut azero_events_receiver: mpsc::Receiver<AzeroMostEvents>,
+        circuit_breaker_sender: broadcast::Sender<CircuitBreakerEvent>,
+        mut circuit_breaker_receiver: broadcast::Receiver<CircuitBreakerEvent>,
+    ) -> Result<CircuitBreakerEvent, AlephZeroEventsHandlerError> {
+        loop {
+            debug!("Ping");
```

Comment:
If we want to keep these kind of logs, it's better to write more, because currently there are more places which log this exact same message.

Reply:
they have the same message, but along that message they log the target (defaults to the module name), and that could be useful in checking if some process is e.g. deadlocked or refuses to die and blocks the restart, these types of problems